### PR TITLE
[Draft] IOFile ReadStream support

### PR DESCRIPTION
**Pull request by elliotsayes**

### CODE REVIEW

File: lib/src/web/web_io.dart [hunk 1]
```diff
@@ -88,37 +89,54 @@ class WebFileSystemProvider implements MultiFileProvider {
     return _ioFolderAdapter.fromIOFiles(files);
   }
 
+  Future<IOFile> _platformFileToStreamFile(PlatformFile platformFile) {
+    return _ioFileAdapter.fromReadStreamGenerator(
+      platformFile.readStream!,
+      platformFile.size,
+      name: platformFile.name,
+      lastModifiedDate: platformFile.lastModified ?? DateTime.now(),
```
Comment (karlprieb) [inline lib/src/web/web_io.dart:97]: For now, let's just use `DateTime.now()` as it was before.
`platformFile.lastModified` breaks on Firefox and we need to be consistent between browsers and platforms.

File: lib/src/io_file.dart [hunk 2]
```diff
@@ -241,6 +321,8 @@ class _FromXFile implements IOFile {
 
   final XFile _file;
 
+  final int _readStreamChunkSize = 256 * 1024;
```
Comment (karlprieb) [inline lib/src/io_file.dart:324]: as we use `256 * 1024` twice, can we use a constant for it?

File: pubspec.yaml [hunk 3]
```diff
@@ -11,7 +11,10 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-  file_picker: ^5.2.4
+  file_picker:
+    git: 
+      url: https://github.com/elliotsayes/flutter_file_picker
```
[the remark this answers is not in the document]
Comment (elliotsayes) [reply pubspec.yaml:16]: Could you tell me where this was merged? There is still a diff with the official repo https://github.com/miguelpruivo/flutter_file_picker/compare/master...elliotsayes:flutter_file_picker:power-stream

File: lib/src/web/web_io.dart (hunk 1 above)
Comment (elliotsayes) [reply lib/src/web/web_io.dart]: Sure thing

File: lib/src/io_file.dart (hunk 2 above)
Comment (elliotsayes) [reply lib/src/io_file.dart]: Good catch!

File: pubspec.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (elliotsayes) [reply pubspec.yaml:16]: PR is here: https://github.com/miguelpruivo/flutter_file_picker/pull/1217
I'm not sure if they will accept it as it changes the API quite heavily, and tbh is quite a niche use case. Let's wait and see

**Comment by karlprieb** [comment]
PR looks good to me. Let's wait until Fri morning for your file_picker PR to be merged so you can update `pubspec` and we can merge this PR. If your PR don't get merged until Fri we can create an ar-io fork and merge it there.

**Pull request merged**